File: asssignment/array/maxProfit/maxprofit.cpp
#include<iostream>
#include<vector>

int maxProfit(const std::vector<int>& prices) {
    if (prices.size() < 2) {
        return 0; // Not enough prices to make a profit
    }

    int buy = prices[0];
    int max_profit = 0;

    for (int price : prices) {
        if (price < buy) {
            buy = price;
        } else {
            int profit = price - buy;
            max_profit = std::max(max_profit, profit);
        }
    }

    return max_profit;
}

int main() {
    std::vector<int> prices = {7,1,5,3,6,4};
    maxProfit(prices);
}

File: asssignment/array/maxProfit/maxprofit.php
<?php

function maxProfit($prices) 
{
    if (count($prices) < 2) {
        return 0; // Not enough prices to make a profit
    }

    $buy = $prices[0];
    $max_profit = 0;

    foreach ($prices as $price) {
        if ($price < $buy) {
            $buy = $price;
        } else {
            $profit = $price - $buy;
            if ($profit > $max_profit) {
                $max_profit = $profit;
            }
        }
    }

    return $max_profit;
}

maxProfit([7,1,5,3,6,4]);